### PR TITLE
Scripts to build java and cxx docker images from local (uncommitted) cod...

### PR DESCRIPTION
**Pull request by donnadionne**
...e and

run interop tests in GCE VMs.

### CODE REVIEW

File: tools/gce_setup/private_build_and_test.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Arguments
+LANGUAGE=$1
+ENV=$2
+TEST=$3
+GIT=$4
+SERVER=${5:-"grpc-docker-server"}
+
+sudo docker run --name="private_images" -v $4:/var/local/git-clone grpc/$1 /var/local/git-clone/grpc/tools/dockerfile/grpc_$1/build.sh
+
+sudo docker commit -m "private image" -a "donnadionne" private_images grpc/private_images
```
Comment (tbetbetbe) [inline tools/gce_setup/private_build_and_test.sh]: does this only work for you at the moment ?

Comment (ctiller) [inline tools/gce_setup/private_build_and_test.sh]: do we need a $USER?

File: tools/gce_setup/private_build_and_test.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+sudo docker tag -f grpc/private_images 0.0.0.0:5000/grpc/private_images
+
+sudo docker push 0.0.0.0:5000/grpc/private_images
+
+sudo docker rmi -f grpc/private_images
+
+sudo docker rm private_images
+
+gcloud compute --project "stoked-keyword-656" ssh --zone "asia-east1-a" "grpc-docker-testclients1" --command "sudo docker pull 0.0.0.0:5000/grpc/private_images"
```
Comment (ctiller) [inline tools/gce_setup/private_build_and_test.sh]: any way to use the default project?

Comment (donnadionne) [inline tools/gce_setup/private_build_and_test.sh]: This script should work for everyone who has docker set up and docker-registry running.  I have a document for that.

I did update to use $USER for committing the image 

Craig I am not sure what you mean by default project.  With this script we will be using the same project as our continuous runs but on dedicated private testing vms.   The plan is that if you change only the client or server side, then you can be testing with server or client side that only has official images.  And all VMs are in the same project.

Comment (nicolasnoble) [inline tools/gce_setup/private_build_and_test.sh]: I think Craig is asking about people who aren't Googlers and who might want to use that script :-)

Comment (donnadionne) [inline tools/gce_setup/private_build_and_test.sh]: I see.

No people who are not a part of our project will not be able to use the script.

For them the plan is provide a recipe.  For example, include instructions like they would need a cloud environment where they have vms, some sort of a storage system, etc...  and then they can take the script, make some modifications or pass in some arguments to make it work.

Comment (ctiller) [inline tools/gce_setup/private_build_and_test.sh]: Let's parameterize the project name to start with at least -- $PROJECT_NAME
perhaps, from a command line arg?

Comment (donnadionne) [inline tools/gce_setup/private_build_and_test.sh]: sure, added more command line args and default values.
PTAL thank!

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request merged**